### DISCUSSION TITLE
Detecting clicks?

### DISCUSSION THREAD

**Discussion by dooovid** [LaunchMenu/LaunchMenu#174] [Ideas]
Is there a way to detect clicks? If not, will this ever be implemented?

**Comment by sancarn** [comment]
Can you clarify, where you'd want to detect clicks?

* Are we talking about clicks within LaunchMenu?
* Are we talking about clicks in other desktop applications?

**Reply by dooovid** [reply]
I am talking about other desktop applications.

**Reply by sancarn** [reply]
We will likely expose that as part of the autocomplete system, yes. However this is more of a question for https://github.com/LaunchMenu/node-global-key-listener or https://github.com/LaunchMenu/OSIntegrationServer

**Reply by dooovid** [reply]
Ah ok I see yes I have been looking for a way to detect keystrokes and clicks globally in node and iohook doesn't work on windows.

**Reply by sancarn** [reply]
I see, you could probably modify `node-global-key-listener` for your needs as at the moment progress on this has largely been halted due to being busy IRL 😊 . Would be happy to accept a PR in regards to that, as it should be as easy as detecting WM_MOUSEDOWN messages

**Reply by dooovid** [reply]
I think I will do a pr for mouse clicks thanks for the help and suggestion!
